### ISSUE TITLE
Indigo 7.5 & Big Sur

### ISSUE THREAD

**Issue by jltnol** [IndigoDomotics/Beacon#3]
So I understand this plugin has been abandon, so doubt anyone will see this, read this, or otherwise take any action. I've used to plugin with indigo for years and it has been rock steady and has worked fine. 

I just recently upgrade my computer that runs indigo, and now the plugin doesn't work. 1st error is there are no configurations in the plugin menu. As I recall, there was a way for new Geofences to auto-populate as devices in Indigo. That doesn't happen any more. :(.  And when current Geofences are crossed, the message from my iPhone is  "FAILED . Could not connect to the server".  I'm positive this isn't a plumbing issue with my set up as nothing in my router has changed and the IP address of the Indigo server on my LAN hasn't changed.   It was working before... just not working now at all.  Hoping someone might be able to take a look... my fingers are crossed.

**Comment by FlyingDiver** [comment]
When you updated your computer, did you update your Geohopper app with the IP address of the new computer?  Or update your DDNS and port forwarding?

**Comment by jltnol** [comment]
Hey Joe

So I have a “personal” URL that I use with both Indigo and my Synology server here at home. The router has always been set to forward the traffic from Geofency (port 6192) to the LAN IP address used for my Indigo server. That IP address is handed out by my router based on the MAC address of the computer, which has been the same, so no real change in the my public IP address, the port forwarding in the router, or the IP address of the Mac running Indigo. And on a hunch, I also changed the URL in Geofency to my public IP address, but still no go.

I’m actually using Geofency not Geohopper but nothing on the app should need changing, as the URL and port # I was using before is exactly the same as now. 

As I recall, the plugin had SOME degree of configuration, probably port # and 1 or 2 other things, and now that config menu is gone… just not there. I even went as far as installing a super squeaky clean version of BigSur, Indigo 7.5, and the plugin on a separate drive as a test, but still no config menu, and no communication between the iOS app and Indigo.

My only other thought was that my ISP was now blocking port 6192, so set up Indigo to use that port of the Indigo iOS Touch app. I disconnected from my home WiFI so was connected from the cell network, and it was working perfectly on port 6192, so I don’t think this issue has anything to do with network setup, but just can’t think of what is so different now than before. 

 Jeffrey

**Comment by FlyingDiver** [comment]
I'm just trying to think of things that could change when you upgraded the computer to Big Sur.  It's the same computer, right?  Just the OS upgrade?

You're saying that you don't see a "Configure..." item in the Plugin menu?  At this point, I would delete the plugin and reinstall.

**Comment by jltnol** [comment]
Same computer, but I was already running Big Sur, and was having some other issues, so wiped the drive and re-installed Big Sur from scratch…. So all new software, but same computer and same networking… )

**Comment by FlyingDiver** [comment]
Did you restore Indigo from backup?  If so, run the installer again to make sure all the permissions are set right.

**Comment by jltnol** [comment]
Yep.  It has recently come to my attention that cloning programs don’t do the best job with permission, so did re-install Indigo, Still not working.  I also deleted the plugin and the plugin permissions in the library folder, re-started, re-installed the plugin, but still no config menu in the plugin.

 And as I’ve mentioned, just to be sure I did a test install on a external drive of BigSur and Indigo, and sill no config menu item…

I’m honestly at a loss...

 JT

**Comment by FlyingDiver** [comment]
No errors in the log?

**Comment by jltnol** [comment]
So if the plugin is working correctly, incoming Port 6192 traffic is intercepted by the plugin and actions in Indigo are performed. 

But since the plugin isn’t working properly, traffic on port 6192 is basically being ignored since the malfunctioning plugin  isn’t listening on that port…and hence… nothing in the log.

 JT

**Comment by FlyingDiver** [comment]
Do you get the startup and shutdown messages?

```
   Starting plugin "Beacon 1.5.3" (pid 98688)
   Started plugin "Beacon 1.5.3"
   Beacon                          Listening on TCP port 6192
   Disabling plugin "Beacon 1.5.3"
   Stopping plugin "Beacon 1.5.3" (pid 98688)
   Stopped plugin "Beacon 1.5.3"
```

**Comment by jltnol** [comment]
No
This is all I get when I reload the plugin:

May 22, 2021 at 8:18:30 PM
   Reloading plugin "Beacon 1.5.3"

**Comment by jltnol** [comment]
So just out of curiosity, when you click on the plugin in the menu, my guess is your dropdown menu looks very different than this one:



> 
> No
> This is all I get when I reload the plugin:
> 
> May 22, 2021 at 8:18:30 PM
>    Reloading plugin "Beacon 1.5.3"
> 
> 
> 
> 
>

**Comment by FlyingDiver** [comment]
Can't do attachments via email.  You need to do them directly on GitHub.

**Comment by jltnol** [comment]
So honestly, I think this IS really some kind of permission problem. I upgraded Indgio thru several OS's, and it was fine I suspected because the previous permissions just came along for the ride. But with a new install of Big Sur, Indigo, and the plugin... and nothing else, all I get in the plugin menu is this:
<img width="472" alt="Screen Shot 2021-05-22 at 8 20 00 PM" src="https://user-images.githubusercontent.com/1592446/119245126-5dbc8580-bb3c-11eb-8591-e8c40267d628.png">

**Comment by FlyingDiver** [comment]
Yeah, it's like it's missing files or they can't be read properly.

**Comment by jltnol** [comment]
For sure. I'm guessing something with the more modern OS's changed in what was required permission wise from plugins, and since this one hasn't been updated since 2017, the installed permissions were never changed.... or at least something like that....  And I did poke around in the Library/Application Support/Perceptive Automation folder in my virgin testbed and found that the permissions for the only other plugin I installed (Airfoil) was different than for Beacon, so I changed the permissions on the Becon plugin, re-started, but still nothing.

This was how permissions were set for the 2 plugins... Airfoil is on the left... but again... after changing Becon to match, so nothing....

<img width="524" alt="Screen Shot 2021-05-22 at 8 39 19 PM" src="https://user-images.githubusercontent.com/1592446/119245362-7037be80-bb3e-11eb-85cc-2e174e1f578d.png">

**Comment by FlyingDiver** [comment]
Finder permissions are worthless.

Open the Terminal app, cd to the Plugins directory (you can type "cd " then drag the Plugins folder from the Finder to the Terminal window.  Then type in "ls -lR".  You'll get something like:

```
./Beacon.indigoPlugin:
total 0
drwxr-xr-x@ 5 jkeenan  staff  160 Dec  8  2017 Contents

./Beacon.indigoPlugin/Contents:
total 8
-rwxr-xr-x@ 1 jkeenan  staff  710 Dec  8  2017 Info.plist
drwxr-xr-x@ 3 jkeenan  staff   96 Dec  8  2017 Resources
drwxr-xr-x@ 6 jkeenan  staff  192 Dec  8  2017 Server plugin

./Beacon.indigoPlugin/Contents/Resources:
total 40
-rw-r--r--@ 1 jkeenan  staff  17556 Dec  8  2017 icon.png

./Beacon.indigoPlugin/Contents/Server plugin:
total 72
-rwxr-xr-x@ 1 jkeenan  staff   1538 Dec  8  2017 Devices.xml
-rwxr-xr-x@ 1 jkeenan  staff   1801 Dec  8  2017 Events.xml
-rwxr-xr-x@ 1 jkeenan  staff   4925 Dec  8  2017 PluginConfig.xml
-rwxr-xr-x@ 1 jkeenan  staff  16949 Dec  8  2017 plugin.py
```

**Comment by jltnol** [comment]
Also installed the iTunes plugin.. Here's what I got:

Administrators-iMac:Plugins admin$ ls -lr
total 0
drwxrwxr-x  3 admin  admin  96 Sep 27  2018 iTunes.indigoPlugin
drwxrwxr-x@ 3 admin  admin  96 Dec  8  2017 Beacon.indigoPlugin
drwxrwxr-x  3 admin  admin  96 Sep 27  2018 Airfoil Pro.indigoPlugin
Administrators-iMac:Plugins admin$ 



Guessing something is wrong with the permissions for Beacon....


On May 22, 2021 at 8:49 PM, Joe Keenan ***@***.***> wrote:


Finder permissions are worthless.
Open the Terminal app, cd to the Plugins directory (you can type "cd " then drag the Plugins folder from the Finder to the Terminal window. Then type in "ls -lR". You'll get something like:
jkeenan@Kraken Plugins % ls -lR
total 0
drwxrwxr-x  3 jkeenan  admin  96 Feb  7  2020 Airfoil Pro.indigoPlugin
drwxr-xr-x@ 3 jkeenan  staff  96 Dec  8  2017 Beacon.indigoPlugin
drwxrwxr-x  3 jkeenan  admin  96 May 18 19:02 Virtual Devices.indigoPlugin

./Airfoil Pro.indigoPlugin:
total 0
drwxrwxr-x  5 jkeenan  admin  160 May 20 12:48 Contents

./Airfoil Pro.indigoPlugin/Contents:
total 8
-rw-rw-r--   1 jkeenan  admin  799 May 18 19:02 Info.plist
drwxrwxr-x   3 jkeenan  admin   96 May 20 12:48 Resources
drwxrwxr-x  13 jkeenan  admin  416 May 22 21:46 Server Plugin

./Airfoil Pro.indigoPlugin/Contents/Resources:
total 16
-rw-rw-r--  1 jkeenan  admin  7087 May 18 19:02 icon.png

./Airfoil Pro.indigoPlugin/Contents/Server Plugin:
total 1816
-rw-rw-r--  1 jkeenan  admin    4440 May 18 19:02 Actions.xml
-rw-rw-r--  1 jkeenan  admin    8942 May 18 19:02 Devices.xml
-rw-rw-r--  1 jkeenan  admin    1025 May 18 19:02 MenuItems.xml
-rwxrwxr-x  1 jkeenan  admin  153616 May 18 19:02 browseBonjour.so
-rwxrwxr-x  1 jkeenan  admin  153552 May 18 19:02 const.so
drwxrwxr-x  3 jkeenan  admin      96 May 20 12:48 images
-rwxrwxr-x  1 jkeenan  admin  153552 May 18 19:02 message.so
-rwxrwxr-x  1 jkeenan  admin  153552 May 18 19:02 plugin.so
-rw-rw-r--  1 jkeenan  admin   63544 May 18 19:02 pybonjour.py
-rw-r--r--  1 jkeenan  admin   59721 May 22 21:46 pybonjour.pyc
-rwxrwxr-x  1 jkeenan  admin  153536 May 18 19:02 ssr.so

./Airfoil Pro.indigoPlugin/Contents/Server Plugin/images:
total 32
-rw-rw-r--  1 jkeenan  admin  16207 May 18 19:02 noimage.png

./Beacon.indigoPlugin:
total 0
drwxr-xr-x@ 5 jkeenan  staff  160 Dec  8  2017 Contents

./Beacon.indigoPlugin/Contents:
total 8
-rwxr-xr-x@ 1 jkeenan  staff  710 Dec  8  2017 Info.plist
drwxr-xr-x@ 3 jkeenan  staff   96 Dec  8  2017 Resources
drwxr-xr-x@ 6 jkeenan  staff  192 Dec  8  2017 Server plugin

./Beacon.indigoPlugin/Contents/Resources:
total 40
-rw-r--r--@ 1 jkeenan  staff  17556 Dec  8  2017 icon.png

./Beacon.indigoPlugin/Contents/Server plugin:
total 72
-rwxr-xr-x@ 1 jkeenan  staff   1538 Dec  8  2017 Devices.xml
-rwxr-xr-x@ 1 jkeenan  staff   1801 Dec  8  2017 Events.xml
-rwxr-xr-x@ 1 jkeenan  staff   4925 Dec  8  2017 PluginConfig.xml
-rwxr-xr-x@ 1 jkeenan  staff  16949 Dec  8  2017 plugin.py

./Virtual Devices.indigoPlugin:
total 0
drwxrwxr-x  4 jkeenan  admin  128 May 18 19:02 Contents

./Virtual Devices.indigoPlugin/Contents:
total 8
-rw-rw-r--  1 jkeenan  admin  875 May 18 19:02 Info.plist
drwxrwxr-x  6 jkeenan  admin  192 May 18 19:02 Server Plugin

./Virtual Devices.indigoPlugin/Contents/Server Plugin:
total 344
-rw-rw-r--  1 jkeenan  admin    1381 May 18 19:02 Actions.xml
-rw-rw-r--  1 jkeenan  admin    9754 May 18 19:02 Devices.xml
-rw-rw-r--  1 jkeenan  admin     366 May 18 19:02 MenuItems.xml
-rwxrwxr-x  1 jkeenan  admin  153552 May 18 19:02 plugin.so

**Comment by FlyingDiver** [comment]
Please post that directly to GitHub.  Email messes it up so I can't format it properly.

**Comment by jltnol** [comment]
Sorry.. I'm working between 2 computes... Here's the output:
Administrators-iMac:Plugins admin$ ls -lr
total 0
drwxrwxr-x  3 admin  admin  96 Sep 27  2018 iTunes.indigoPlugin
drwxrwxr-x@ 3 admin  admin  96 Dec  8  2017 Beacon.indigoPlugin
drwxrwxr-x  3 admin  admin  96 Sep 27  2018 Airfoil Pro.indigoPlugin
Administrators-iMac:Plugins admin$

**Comment by FlyingDiver** [comment]
R needs to be capitalized.  "ls -lR".

**Comment by jltnol** [comment]
Administrators-iMac:Plugins admin$ ls -lr
total 0
drwxrwxr-x  3 admin  admin  96 Sep 27  2018 iTunes.indigoPlugin
drwxrwxr-x@ 3 admin  admin  96 Dec  8  2017 Beacon.indigoPlugin
drwxrwxr-x  3 admin  admin  96 Sep 27  2018 Airfoil Pro.indigoPlugin
Administrators-iMac:Plugins admin$ ls -iR
58501 Airfoil Pro.indigoPlugin	58490 Beacon.indigoPlugin	58899 iTunes.indigoPlugin

./Airfoil Pro.indigoPlugin:
58502 Contents

./Airfoil Pro.indigoPlugin/Contents:
58390 Info.plist	58516 Resources		58503 Server Plugin

./Airfoil Pro.indigoPlugin/Contents/Resources:
58389 icon.png

./Airfoil Pro.indigoPlugin/Contents/Server Plugin:
58379 Actions.xml	58377 browseBonjour.so	58387 message.so	64608 pybonjour.pyc
58382 Devices.xml	58385 const.so		58378 plugin.so		58386 ssr.so
58383 MenuItems.xml	58506 images		58384 pybonjour.py

./Airfoil Pro.indigoPlugin/Contents/Server Plugin/images:
58381 noimage.png

./Beacon.indigoPlugin:
58491 Contents

./Beacon.indigoPlugin/Contents:
58492 Info.plist	58493 Resources		58495 Server plugin

./Beacon.indigoPlugin/Contents/Resources:
58494 icon.png

./Beacon.indigoPlugin/Contents/Server plugin:
58498 Devices.xml	58497 Events.xml	58496 PluginConfig.xml	58499 plugin.py

./iTunes.indigoPlugin:
58900 Contents

./iTunes.indigoPlugin/Contents:
58453 Info.plist	58901 Server Plugin

./iTunes.indigoPlugin/Contents/Server Plugin:
58443 Actions.xml	58444 PluginConfig.xml	58448 itunes.so		58447 pybonjour.py
58445 Devices.xml	58449 command.so	58451 pairingThread.so	66849 pybonjour.pyc
58446 MenuItems.xml	58450 dacp.so		58442 plugin.so		58452 statusThread.so
Administrators-iMac:Plugins admin$

**Comment by FlyingDiver** [comment]
That time you put in an i instead of an l.

**Comment by jltnol** [comment]
```
Administrators-iMac:Plugins admin$ ls -lR


./Beacon.indigoPlugin:
total 0
drwxr-xr-x@ 5 admin  admin  160 May 22 19:08 Contents

./Beacon.indigoPlugin/Contents:
total 8
-rwxr-xr-x@ 1 admin  admin  710 Dec  8  2017 Info.plist
drwxr-xr-x@ 3 admin  admin   96 Dec  8  2017 Resources
drwxr-xr-x@ 6 admin  admin  192 Dec  8  2017 Server plugin

./Beacon.indigoPlugin/Contents/Resources:
total 40
-rw-r--r--@ 1 admin  admin  17556 Dec  8  2017 icon.png

./Beacon.indigoPlugin/Contents/Server plugin:
total 72
-rwxr-xr-x@ 1 admin  admin   1538 Dec  8  2017 Devices.xml
-rwxr-xr-x@ 1 admin  admin   1801 Dec  8  2017 Events.xml
-rwxr-xr-x@ 1 admin  admin   4925 Dec  8  2017 PluginConfig.xml
-rwxr-xr-x@ 1 admin  admin  16949 Dec  8  2017 plugin.py


```

**Comment by FlyingDiver** [comment]
What user account on a Mac has username admin, group admin?  That's not the standard user account.

**Comment by jltnol** [comment]
When I set up a new computer,  I usually put in an admin account so in case something goes wrong with my normal user account, I can get back in and try to fix it.  when I set this up as my test bed, I just didn't set up a 2nd user account. My normal Indigo user account is different.... but just not on this computer...

**Comment by jltnol** [comment]
So not sure I really answered your question... The user account is Administrator... short name = admin.

**Comment by FlyingDiver** [comment]
The permissions all look right, even if the username and group are not what I would have expected.  I'm out of ideas.

**Comment by jltnol** [comment]
Ok soooo that output was from my "Test" computer... if you like, I can output the same from my "real" Indigo computer... not that I'd expect anything different......since they are both doing exactly the same thing....

**Comment by FlyingDiver** [comment]
OOps

**Comment by jltnol** [comment]
so the output from my real Indigo Server is too long to post  in one comment..but here is the 1st part  thru the Beacon plugin info:



```
Kitchen-Automation:Plugins kitchen$ ls -lR
total 0
drwxrwxr-x  3 kitchen  admin   96 Dec 22 17:02 Airfoil Pro.indigoPlugin
drwxr-xr-x@ 3 kitchen  admin   96 Dec  8  2017 Beacon.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 May 23  2017 BetterEmail.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Apr 25  2018 ClockDisplay.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Aug  1  2019 Cynical Behaviors.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Dec 22  2016 Cynical Denon.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Dec 30  2017 DSC Alarm.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Jul 18  2018 EPS HomeKit Bridge.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Aug 26  2016 EPS Mac Commander.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Aug 28  2016 EPS Scene Toggle.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Jul  3  2018 EPS Super Conditions.indigoPlugin
drwxrwxr-x  3 kitchen  admin   96 Dec 22 17:02 Global Property Manager.indigoPlugin
drwxrwxrwx@ 4 kitchen  admin  128 Dec  9  2017 Hue Lights.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Jun 25  2020 LEDeffects.indigoPlugin
drwxrwxrwx@ 4 kitchen  admin  128 Dec 27 14:10 LIFX.indigoPlugin
drwxrwxr-x  3 kitchen  admin   96 Dec 22 17:02 NOAA Weather.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Apr 27  2020 Timed Devices.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Aug 19  2017 Twilio.indigoPlugin
drwxrwxr-x  3 kitchen  admin   96 Dec 22 17:02 Virtual Devices.indigoPlugin
drwxrwxrwx@ 4 kitchen  admin  128 Feb 28 12:16 WeatherFlow.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 May  7  2019 Yahama RX.indigoPlugin
drwxrwxrwx@ 4 kitchen  admin  128 Oct 28  2017 ZwaveLockManager.indigoPlugin
drwxrwxrwx@ 4 kitchen  admin  128 Dec 17  2016 iTunes Local Control.indigoPlugin
drwxrwxr-x  3 kitchen  admin   96 Dec 22 17:02 iTunes.indigoPlugin
drwxrwxrwx@ 5 kitchen  admin  160 Feb  5 09:25 matplotlib.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Jul 19  2020 nodeMatrix.indigoPlugin
drwxrwxrwx@ 3 kitchen  admin   96 Jul  1  2014 survey2.indigoPlugin
drwxrwxrwx@ 4 kitchen  admin  128 Jun 24  2020 utilities.indigoPlugin

./Airfoil Pro.indigoPlugin:
total 0
drwxrwxr-x  5 kitchen  admin  160 Dec 22 17:02 Contents

./Airfoil Pro.indigoPlugin/Contents:
total 8
-rw-rw-r--   1 kitchen  admin  796 Dec 22 17:02 Info.plist
drwxrwxr-x   3 kitchen  admin   96 Dec 22 17:02 Resources
drwxrwxr-x  13 kitchen  admin  416 May 20 19:29 Server Plugin

./Airfoil Pro.indigoPlugin/Contents/Resources:
total 16
-rw-rw-r--  1 kitchen  admin  7087 Dec 22 17:02 icon.png

./Airfoil Pro.indigoPlugin/Contents/Server Plugin:
total 688
-rw-rw-r--  1 kitchen  admin   4440 Dec 22 17:02 Actions.xml
-rw-rw-r--  1 kitchen  admin   8942 Dec 22 17:02 Devices.xml
-rw-rw-r--  1 kitchen  admin   1025 Dec 22 17:02 MenuItems.xml
-rwxrwxr-x  1 kitchen  admin  34784 Dec 22 17:02 browseBonjour.so
-rwxrwxr-x  1 kitchen  admin  34704 Dec 22 17:02 const.so
drwxrwxr-x  3 kitchen  admin     96 Dec 22 17:02 images
-rwxrwxr-x  1 kitchen  admin  34736 Dec 22 17:02 message.so
-rwxrwxr-x  1 kitchen  admin  47168 Dec 22 17:02 plugin.so
-rw-rw-r--  1 kitchen  admin  63544 Dec 22 17:02 pybonjour.py
-rw-r--r--  1 kitchen  admin  59520 May 20 19:29 pybonjour.pyc
-rwxrwxr-x  1 kitchen  admin  38832 Dec 22 17:02 ssr.so

./Airfoil Pro.indigoPlugin/Contents/Server Plugin/images:
total 32
-rw-rw-r--  1 kitchen  admin  16207 Dec 22 17:02 noimage.png

./Beacon.indigoPlugin:
total 0
drwxr-xr-x@ 5 kitchen  admin  160 May 20 15:13 Contents

./Beacon.indigoPlugin/Contents:
total 8
-rwxr-xr-x@ 1 kitchen  admin  710 Dec  8  2017 Info.plist
drwxr-xr-x@ 3 kitchen  admin   96 Dec  8  2017 Resources
drwxr-xr-x@ 6 kitchen  admin  192 Dec  8  2017 Server plugin

./Beacon.indigoPlugin/Contents/Resources:
total 40
-rw-r--r--@ 1 kitchen  admin  17556 Dec  8  2017 icon.png

./Beacon.indigoPlugin/Contents/Server plugin:
total 72
-rwxr-xr-x@ 1 kitchen  admin   1538 Dec  8  2017 Devices.xml
-rwxr-xr-x@ 1 kitchen  admin   1801 Dec  8  2017 Events.xml
-rwxr-xr-x@ 1 kitchen  admin   4925 Dec  8  2017 PluginConfig.xml
-rwxr-xr-x@ 1 kitchen  admin  16949 Dec  8  2017 plugin.py
```

**Comment by FlyingDiver** [comment]
I don't see anything wrong there either.  I have no idea what's the problem.  It loads fine on my development system, which is also running Big Sur.

**Comment by jltnol** [comment]
Ok.  I was pretty sure this was going to be a simple fix, which is whey I did the clean OS and Indigo install, to rule out anything else that might have been causing problems.  But... the best laid plans....

I may go back to an earlier version of the OS, try to get it running, and then upgrade to BigSur.  

Thanks for your help and time looking into this. I appreciate it.

**Comment by jltnol** [comment]
Sunday Morning Update:  I did finally get this to work but it took some doing. I've got an older MacPro laying around which has Sierra on it. I installed Indigo 7.5, and the Beacon plugin. Presto!  Working as it should. I used Carbon Copy Cloner to clone this drive to an external one, booted from the external drive, and Beacon is still working. Next, I attached my USB Clone drive to the iMac that needs to be running Indigo. Booted from the external drive, used Disc Utility to fully reformat the internal drive, used CCC to clone my working USB drive to the internal SSD drive, booted from the internal drive, and Beacon is still working. All good news. At this point, I'm still using the default Indigo database and just the Beacon Plugin. Next up, upgrading the iMac to Big Sur. Done. Beacon still working with the default database on BigSur. Next I coped the right folders over from a previous install of Indigo, including my working database file, and Beacon is still working.  So now I have Indigo 7.5 running on Big Sur and a working Beacon plugin.  I am happy, but still very curious.

So the story doesn't end here. Since this OS was migrated from a much older computer, it has a lot of cruft and junk on it, including MS office 2008. So in an effort to keep my Indigo Server as light as possible, I planned to start with a fresh install of Sierra, since I knew at least things were working on it.    I actually have a Sierra Install app laying around so I copied the install app to a totally separate laptop, plugged in an external USB drive, reformatted the external USB drive, and installed Sierra on it. After booting from the newly installed copy of Sierra, then I installed Indigo 7.5 and the Beacon plugin. And guess what?  I didn't work. Same problem. No Config menu item.  I did remember that Indigo needs  extended access, so went to the Security & Privacy pane, and to Accessibility  pane, added, Indigo, re-started, re-installed Indigo 7.5, and STILL the Beacon plugin didn't work. 

Then it also occurred to me that my "test os", a clean Apple Internet Restore of Big Sur I was using last night never asked me to grant Indigo Full Disc Access, so that may have been the issue last night.  But that doesn't explain my issue with a clean install of Sierra today. 

Clearly, I must be doing something wrong...  I've been very meticulous when doing these test OS installs... trying to keep things as simple and uncomplicated as possible to rule out any interference from whatever might be/ could be a conflict... And I've used 2 separate USB drives, all of which were totally erased and re-formatted before my Test OS installs... so again... no answers... just questions. 

So probably safe to mark this Closed....but I'd be interested for others to follow my path, do a fresh install of Big Sur, Indigo 7.5, and the Plugin, just to see if the problem I had could be replicated.
